### PR TITLE
Prevent TouchCursor from being cleared while it is still needed.

### PR DESCRIPTION
**Pull request by a-gagnon**
Fix an issue with mobile devices where the TouchCursor would disappear for no apparent reason. It it due to the order in which we call enableSnap/enableLocate. Previously, only enableSnap would determine the presence of the TouchCursor. Now, both enableSnap and enableLocate do (see https://github.com/imodeljs/imodeljs/pull/1038).

The following sequence will lead to the problem:
  ToolSettings.enableVirtualCursorForLocate = true;

  // Tool starts. isElementSelected = false
  accuSnap.enableLocate(!isElementSelected); // on
  accuSnap.enableSnap(isElementSelected); // off
  
  // User is prompted to tap to invoke the TouchCursor
  **user taps and invokes TouchCursor**
  **user taps and selects an element**
  
  accuSnap.enableLocate(!isElementSelected); // off. Clears the TouchCursor.
  accuSnap.enableSnap(isElementSelected); // on.
  // User is prompted to tap to invoke the TouchCursor
  
Note: tools that directly call enableSnap/enableLocate might still experience this problem. Fixing this problem entirely would require AccuSnap to wait some time (1 frame?) before effectively clearing the TouchCursor.

### CODE REVIEW

**Review by pmconne — COMMENTED** [review]
I defer to @bbastings.

File: core/frontend/src/tools/Tool.ts [hunk 1]
```diff
@@ -646,11 +646,14 @@ export abstract class InteractiveTool extends Tool {
       toolAdmin.setLocateCursor(enableLocate);
     }
 
-    accuSnap.enableLocate(enableLocate);
-    if (undefined !== enableSnap)
-      accuSnap.enableSnap(enableSnap);
-    else
-      accuSnap.enableSnap(false);
+    // Always set the one that is true first, otherwise AccuSnap will clear the TouchCursor for no apparent reason.
```
Comment (bbastings) [inline core/frontend/src/tools/Tool.ts:649]: Not going to approve with this "no apparent reason" comment, there's obviously a reason.

Comment (a-gagnon) [reply core/frontend/src/tools/Tool.ts:649]: Right... There is. As described in the conversation, the TouchCursor isn't only invoked by enabling snap anymore. When `ToolSettings.enableVirtualCursorForLocate=true`, enabling locate will invoke it as well. If both locate/snap end up being turned off at the same time, the TouchCursor gets cleared, even if the next statement is to turn one of them back on. I've updated the example in the original post.

Comment (bbastings) [reply core/frontend/src/tools/Tool.ts]: Right, and it's not true that a tool is always snapping or locating. The select tool for examples, pick a few elements using the touch cursor, then change mode to line or box select...not neither locate or snap is enabled and the touch cursor needs to go away. This change should be fine, any tool that enables locate really needs to call this method as it does a lot of things you'd get wrong otherswise, i.e. cursor, locate circle, coord lock overrides, etc.

**Review by bbastings — APPROVED** [review]

**Comment by a-gagnon** [comment]
@kabentley Could you please have a look? I need to provide updates once in a while for a fix that depends on this one.

**Review by kabentley — APPROVED** [review]

**Comment by calebmshafer** [comment]
@a-gagnon can you please run `rush change`? Looks like the CI job is failing due to that.

**Comment by a-gagnon** [comment]
@calebmshafer Done, but it looks like every reviewer from every repo needs to accept changes because of the `rush change` output...

**Comment by pmconne** [comment]
> @calebmshafer Done, but it looks like every reviewer from every repo needs to accept changes because of the `rush change` output...

@calebmshafer is that because he's running it on a fork?

**Comment by calebmshafer** [comment]
> > @calebmshafer Done, but it looks like every reviewer from every repo needs to accept changes because of the `rush change` output...
> 
> @calebmshafer is that because he's running it on a fork?

@pmconne yes it looks like `rush change` ran against the master branch of @a-gagnon's fork rather than the main one. It should automatically find the `imodeljs:master` though... Let me try it and see.

**Pull request merged**